client/src/components/modal-forms/form-components/EditRosTemplate.js

```diff
@@ -35,30 +35,25 @@ export const EditRosTemplate = ({
 
   return (
     <>
-      <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onSubmit}>
-        {({ validateForm, submitForm, values, setSubmitting }) => (
-          <FormikForm>
-            <Box mb={2}>
-              {children}
+      <Box mb={2}>
+        {children}
+        <Divider className={classes.formDivider} />
 
-              <Divider className={classes.formDivider} />
-
-              <FormButtons
-                onClose={onClose}
-                onSubmit={async () => {
-                  setSubmitting(false);
-                  const res = await validateForm();
-                  if (Object.entries(res)?.length === 0) {
-                    openConfirmationDialog(values);
-                    return;
-                  }
-                  await submitForm();
-                }}
-              />
-            </Box>
-          </FormikForm>
-        )}
-      </Formik>
+        <FormButtons
+          onClose={onClose}
+          onSubmit={async () => {
```

Comment:
Just looking at this again, if the form button was `type='submit'` I imagine you'd be able to skip a lot of this extra submission/validation logic

Reply:
If I make button `type='submit'`, it'll trigger the submission and won't show the confirmation modal even if I set `setSubmitting(false);`

Reply:
You would have to modify the `onSubmit` logic to trigger the confirmation modal and then let the confirmation modal handle the network request